.github/workflows/build_and_test.yml

```diff
@@ -169,19 +169,20 @@ jobs:
 
   formatting-check:
     name: "C++ format check"
-    runs-on: ubuntu-20.04
-    strategy:
-      matrix:
-        path:
-          - 'c++'
-          - 'tools'
+    runs-on: ubuntu-latest
```

Comment:
nit: It would be better to use a fixed version. Otherwise, a breaking upgrade may result in CI failures, e.g. unable to find a specific version of clang tools.